yt/fields/field_type_container.py

```diff
@@ -0,0 +1,137 @@
+"""
+A proxy object for field descriptors, usually living as ds.fields.
+"""
+
+import weakref
+import textwrap
+import inspect
+from yt.extern.six import string_types
+from yt.fields.derived_field import \
+    DerivedField
+
+def _fill_values(values):
+    value = '<div class="rendered_html jp-RenderedHTMLCommon">' + \
+            '<table><thead><tr><th>Name</th><th>Type</th>' + \
+            '<th>Value</th></tr></thead><tr><td>' + \
+            '</td></tr><tr><td>'.join(['{0}</td><td>{1}</td><td>{2}'.format(
+                v, type(values[v]).__name__, str(values[v]))
+                for v in sorted(values)]) + \
+            '</td></tr></table></div>'
+    return value
```

Comment:
Would it be crazy to just include the values of closure variables in the output of `DerivedField.get_source()`? One could imagine doing some regex magic to actually edit the code in the field definition to replace variables with their values. That might be too magic though, and brittle...

Reply:
I wouldn't say it's *crazy* per se, but I did not trust myself to do it correctly in this first pass.  Maybe what we could try is have a tab (default) where we do that, and then one where it's the raw source?  I'm not sure.  Maybe there's a way to use the AST or something?  That might be feasible.

Reply:
It may be more straightforward than I thought.  I will give it a go. A `NodeTransformer` might be able to do it without too much trouble.  https://docs.python.org/3.6/library/ast.html#ast.NodeTransformer

This could even be useful in the future for things like field day work.